src/gitops/components/application/ApplicationSetDetailsPage.tsx

```diff
@@ -0,0 +1,636 @@
+                          </div>
+
+                          {/* Annotations Section - matches Console style */}
+                          <div className="pf-v6-c-description-list__group">
+                            <dt className="pf-v6-c-description-list__term" data-test-selector="details-item-label_Annotations" style={{ margin: 0 }}>
+                              <span>Annotations</span>
+                            </dt>
+                            <dd className="pf-v6-c-description-list__description" style={{ padding: 0, marginTop: 0 }}>
+                              <div style={{ display: 'inline-block' }}>
+                                <div style={{ display: 'flex', alignItems: 'center', marginBottom: 4, width: '100%' }}>
+                                  <a
+                                    style={{
+                                      fontSize: 15,
+                                      color: '#73bcf7',
+                                      textDecoration: 'underline',
+                                      cursor: 'pointer',
+                                      display: 'inline-flex',
+                                      alignItems: 'center',
+                                    }}
+                                    tabIndex={0}
+                                    role="button"
+                                    onClick={launchAnnotationsModal}
+                                    onKeyPress={e => { if (e.key === 'Enter' || e.key === ' ') launchAnnotationsModal(); }}
+                                    aria-label="Edit annotations"
+                                  >
+                                    {countAnnotations} annotation{countAnnotations !== 1 ? 's' : ''}
+                                    <PencilAltIcon style={{ marginLeft: 6, fontSize: 15, color: '#73bcf7' }} />
+                                  </a>
+                                </div>
+                              </div>
+                            </dd>
+                          </div>
+
+                          <div className="pf-v6-c-description-list__group">
+                            <dt className="pf-v6-c-description-list__term" data-test-selector="details-item-label_Created">
+                              <div className="pf-v6-l-split pf-v6-u-w-100">
+                                      <Button
+                                        variant="link"
+                                        icon={<PencilAltIcon />}
+                                        onClick={launchLabelsModal}
+                                        style={{
+                                          padding: 0,
+                                          position: 'absolute',
+                                          top: -24,
+                                          right: 0,
+                                          fontSize: 13,
+                                        }}
+                                        aria-label="Edit labels"
```

Comment:
We should enable or disable the button if we know whether this resource can be updated.